### PR TITLE
refactor: clean up Node.js cli arg parsing

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/39288.

More clearly indicates that `--js-flags` is _only_ for V8 flags, as well as cleans up Node.js cli flag parsing.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: shell/common/node_bindings.cc [hunk 1]
```diff
@@ -417,16 +414,7 @@ void NodeBindings::SetNodeCliFlags() {
     args.push_back("--no-experimental-fetch");
   }
 
-  std::vector<std::string> errors;
-  const int exit_code = ProcessGlobalArgs(&args, nullptr, &errors,
-                                          node::kDisallowedInEnvironment);
```
Comment (codebytere) [inline shell/common/node_bindings.cc:422]: This is handled by `node::InitializeNodeWithArgs`

File: shell/app/node_main.cc [hunk 2]
```diff
@@ -62,40 +63,26 @@ int SetNodeCliFlags() {
...
-  const auto argv = base::CommandLine::ForCurrentProcess()->argv();
-  std::vector<std::string> args;
-
   // TODO(codebytere): We need to set the first entry in args to the
   // process name owing to src/node_options-inl.h#L286-L290 but this is
   // redundant and so should be refactored upstream.
-  args.reserve(argv.size() + 1);
-  args.emplace_back("electron");
-
+  std::vector<std::string> args;
```
Comment (ckerr) [inline shell/app/node_main.cc:69]: If we keep this function's current form of returning a vector, the `reserve()` call should be restored here

File: shell/app/node_main.cc [hunk 3]
```diff
@@ -62,40 +63,26 @@ int SetNodeCliFlags() {
...
       // when setting up cli flags and env vars. Since we're outlawing these
-      // flags (making them errors) return 9 here for consistency.
-      return 9;
+      // flags (making them errors) exit with the same error code for
+      // consistency.
+      exit(9);
     } else {
-      args.push_back(option);
+      args.push_back(arg);
     }
```
Comment (ckerr) [inline shell/app/node_main.cc:82]: _(nit)_ Not new to this PR, but the `else {` and `}` lines 80 and 82 are not necessary & can be removed

File: docs/api/command-line-switches.md [hunk 4]
```diff
@@ -116,14 +116,20 @@ Ignore the connections limit for `domains` list separated by `,`.
...
 ```
 
-See the [Node.js documentation][node-cli] or run `node --help` in your terminal for a list of available flags. Additionally, run `node --v8-options` to see a list of flags that specifically refer to Node.js's V8 JavaScript engine.
+Run `node --v8-options` or `electron --js-flags="--help"` in your terminal for the list of available flags.  These can be used to enable early-stage JavaScript features, or log and manipulate garbage collection, among other things.
+
+For example, to trace V8 optimization and deoptimization:
+
+```sh
+$ electron --js-flags="--trace-opt --trace-deopt" your-app
+```
```
Comment (ckerr) [inline docs/api/command-line-switches.md:132]: This is a nice improvement :+1:

File: shell/app/node_main.cc [hunk 5]
```diff
@@ -62,40 +63,26 @@ int SetNodeCliFlags() {
...
-  args.reserve(argv.size() + 1);
-  args.emplace_back("electron");
-
+  std::vector<std::string> args;
   for (const auto& arg : argv) {
-#if BUILDFLAG(IS_WIN)
-    const auto& option = base::WideToUTF8(arg);
-#else
-    const auto& option = arg;
-#endif
```
[the remark this answers is not in the document]
Comment (codebytere) [reply shell/app/node_main.cc:79]: @ckerr i removed the code here https://github.com/electron/electron/pull/39465/files#diff-2fa5dbc7196ffe767cc9a4e7c100c277e895dc6fd8e5287651af22f6681fa724L65 - and now pass an existing `std::string` vector

File: shell/app/node_main.cc [hunk 6]
```diff
@@ -62,40 +62,21 @@ int SetNodeCliFlags() {
...
-    const auto stripped = base::StringPiece(option).substr(0, option.find('='));
-    if (disallowed.contains(stripped)) {
-      LOG(ERROR) << "The Node.js cli flag " << stripped
+    const auto key = base::StringPiece(arg).substr(0, arg.find('='));
```
Comment (zcbenz) [inline shell/app/node_main.cc:69]:
```suggestion
    const auto key = arg.substr(0, arg.find('='));
```

**Review by zcbenz — APPROVED** [review]

File: shell/app/node_main.cc (hunk 6 above)
Comment (ckerr) [reply shell/app/node_main.cc:69]: @zcbenz that's terser, but does cause a new `std::string` to be constructed with a copy of the front of `arg`.

Probably (very slightly) better to use the string view instead?

File: shell/app/node_main.cc [hunk 7]
```diff
@@ -62,40 +62,21 @@ int SetNodeCliFlags() {
       "--use-openssl-ca",
   });
 
-  const auto argv = base::CommandLine::ForCurrentProcess()->argv();
   std::vector<std::string> args;
-
-  // TODO(codebytere): We need to set the first entry in args to the
-  // process name owing to src/node_options-inl.h#L286-L290 but this is
-  // redundant and so should be refactored upstream.
   args.reserve(argv.size() + 1);
```
Comment (ckerr) [inline shell/app/node_main.cc:66]: `args` isn't used anymore; these two lines 65-66 can be removed

**Review by ckerr — APPROVED** [review]

File: shell/app/node_main.cc (hunk 6 above)
Comment (zcbenz) [reply shell/app/node_main.cc:66]: Didn't realize that, I'm good as it is.

**Pull request merged**